indico/modules/events/registration/client/js/form/fields/TimetableSessions.jsx

```diff
@@ -0,0 +1,297 @@
+
+import '../../../styles/regform.module.scss';
+import {getStaticData} from '../selectors';
+
+const sessionDataSchema = PropTypes.arrayOf(
+  PropTypes.shape({
+    fullTitle: PropTypes.string.isRequired,
+    id: PropTypes.number.isRequired,
+    startDt: PropTypes.string.isRequired,
+  })
+);
+
+function TimetableSessionsComponent({
+  value,
+  name,
+  sessionData,
+  onChange,
+  onFocus,
+  onBlur,
+  display,
+  minimum,
+  maximum,
+  ...props
+}) {
+  const [expandedHeaders, setExpandedHeaders] = useState([]);
+
+  const markTouched = () => {
+    onFocus();
+    onBlur();
+  };
+
+  function handleValueChange(e, data) {
+    markTouched();
+    const id = data.value;
+    const newValue = _.xor(value, [id]);
+    onChange(newValue);
+  }
+
+  function handleHeaderClick(e, accordion) {
+    if (expandedHeaders.includes(accordion.index)) {
+      setExpandedHeaders(expandedHeaders.filter(el => el !== accordion.index));
+    } else {
+      setExpandedHeaders([...expandedHeaders, accordion.index]);
+    }
+  }
+
+  useEffect(() => {
+    setExpandedHeaders(
+      sessionData && display === 'expanded' ? [...Array(_.keys(sessionData).length).keys()] : []
+    );
+  }, [sessionData, display]);
+
+  return (
+    <Form.Group styleName="timetablesessions-field">
+      <Accordion styled exclusive={false}>
+        {sessionData &&
+          Object.keys(sessionData).map((date, index) => (
+            <SessionBlockHeader
+              value={value}
+              index={index}
+              name={name}
+              data={sessionData[date]}
+              label={moment(date).format('dddd ll')}
+              isExpanded={expandedHeaders.includes(index)}
+              key={date}
+              handleValueChange={handleValueChange}
+              onChange={onChange}
+              onClick={handleHeaderClick}
+              props={props}
+            />
+          ))}
+      </Accordion>
+    </Form.Group>
+  );
+}
+
+TimetableSessionsComponent.propTypes = {
+  value: PropTypes.arrayOf(PropTypes.number).isRequired,
+  name: PropTypes.string.isRequired,
+  sessionData: PropTypes.objectOf(sessionDataSchema),
+  onChange: PropTypes.func.isRequired,
+  onFocus: PropTypes.func.isRequired,
+  onBlur: PropTypes.func.isRequired,
+  display: PropTypes.oneOf(['expanded', 'collapsed']),
+  maximum: PropTypes.number.isRequired,
+  minimum: PropTypes.number.isRequired,
+};
+
+TimetableSessionsComponent.defaultProps = {
+  sessionData: {},
+  display: 'expanded',
+};
+
+function SessionBlockHeader({
+  value,
+  index,
+  data,
+  label,
+  isExpanded,
+  handleValueChange,
+  onChange,
+  onClick,
+}) {
+  const childrenIds = data.map(e => e.id);
+  const isAllChecked = childrenIds.every(id => value.includes(id));
+  const isAnyChecked = childrenIds.some(id => value.includes(id));
+  const selectedBlocks = value.filter(element => childrenIds.includes(element)).length;
+
+  function handleHeaderChange(e) {
+    e.stopPropagation();
```

Comment:
My above comment was because using `e` for different purposes is strange.

In fact, I'd use `evt` here which is much more common across our codebase.

```suggestion
  function handleHeaderChange(evt) {
    evt.stopPropagation();
```